content/applications/finance/fiscal_localizations/kenya.rst

```diff
@@ -7,114 +7,320 @@ Kenya
+   vendor bill to the KRA.
+#. Once all steps are completed, the vendor bill can be sent by clicking :guilabel:`Send to eTIMS`.
+   When the vendor bill has been confirmed on eTIMS, the **KRA invoice number** can be found under
+   the :guilabel:`eTIMS Details`.
+
+.. image:: kenya/purchase-order-lines.png
+   :alt: Bill registration steps.
+
+Invoicing
+=========
+
+.. note::
+   The KRA does *not* accept sales if the product is not in stock.
+
+This is the **advised sales flow** in Odoo when selling:
+
+#. Create a **sales order**;
+#. :guilabel:`Validate` the invoice;
+#. :guilabel:`Confirm` the invoice;
+#. Click :guilabel:`Send and print`, and then tick the :guilabel:`Send to eTIMS` box.
+#. Finally, click :guilabel:`Send & print` to send the invoice.
+
+Once the invoice has been sent and signed by the KRA, the following information can be found on
+it:
+
+- **KRA invoice number**;
+- Mandatory KRA invoice fields, such as **SCU information**, **date**, **SCU ID**, **receipt
+  number**, **item count**, **internal date**, and **receipt signature**;
+- The **KRA tax table**;
+- A unique **KRA QR code** for the signed invoice.
+
+Imports
+=======
+
+Customs imports codes are **automatically** fetched from the KRA eTIMS API servers **every day**.
+These codes can be **manually** fetched by enabling :doc:`../../general/developer_mode` and going to
+:menuselection:`Settings --> Technical --> Automation: Scheduled Actions --> KE eTIMS: Receive
+Customs Imports from the OSCU`. Here, click :guilabel:`Run Manually` to fetch the codes. Import
+items can be found under :menuselection:`Accounting --> Vendors --> Customs Imports`.
+
+The following steps are required to send and have **customs imports** signed by the KRA:
+
+#. The customs import is fetched automatically from the KRA.
+#. Users *must* match the imported item with an existing registered product (or create a product if
+   no related product exists) in the :guilabel:`Product` field.
+#. Set a vendor in the :guilabel:`Partner` field.
+#. Based on the partner, the user must match the imported item with its related purchase order
+   (see :ref:`purchase steps <kenya/purchases>`). The stock must be correctly adjusted when then
+   customs import is approved.
+#. If no related purchase order exists, the user must create one. The :guilabel:`Product` and
+   :guilabel:`Partner` must have been set before.
+#. Finally, click :guilabel:`Match and Approve` or :guilabel:`Match and Reject`, depending on the
+   situation of the goods.
+
+.. note::
+   The JSON file received from the KRA is attached to the form view of the customs import.
+
+BOM
+===
+
+The KRA requires all BOMs to be sent to them. To send BOMs to eTIMS, the product and its components
+*must* be :ref:`registered <kenya/product-registration>`. To access a product's BOM, click on the
+product, and then click on the :guilabel:`Bill Of Materials`.
+
+Fill in the KRA's required fields, and then click :guilabel:`Send to eTIMS`. The successful
```

Comment:
```suggestion
Fill in the KRA's required fields in the :guilabel:`KRA eTIMS details` section of the :guilabel:`Accounting` tab, and then click :guilabel:`Send to eTIMS`. The successful
```
option 2 would be to refer to where you talk about these fields in the doc (Product registration section), might be better actually. In any case i feel like we need to add something